libparsec/crates/types/src/ext_types.rs

```diff
@@ -85,7 +85,7 @@ impl<'de> serde::de::Visitor<'de> for DateTimeExtVisitor {
         let ts = f64::from_be_bytes(
             data[..F64_SIZE]
                 .try_into()
-                .unwrap_or_else(|_| unreachable!()),
+                .expect("DateTimeExtVisitor `data` should be convertible to a [u8; 8] byte array"),
```

Comment:
The real reason is we already check the boundary on line `79`.

Here you're trying to convert a `&[u8]` to `[u8; 8]` that conversion can fail if `data.len() != 8`, that check if effectively already done above